numpy/linalg/linalg.py

```diff
@@ -69,22 +69,19 @@ class LinAlgError(Exception):
     """
     pass
 
-# Dealing with errors in _umath_linalg
-
-_linalg_error_extobj = None
 
 def _determine_error_states():
-    global _linalg_error_extobj
     errobj = geterrobj()
     bufsize = errobj[0]
 
     with errstate(invalid='call', over='ignore',
                   divide='ignore', under='ignore'):
         invalid_call_errmask = geterrobj()[1]
 
-    _linalg_error_extobj = [bufsize, invalid_call_errmask, None]
+    return [bufsize, invalid_call_errmask, None]
 
-_determine_error_states()
+# Dealing with errors in _umath_linalg
+_linalg_error_extobj = _determine_error_states()
 
```

Comment:
Can we `del _determine_error_states` on the next line, to ensure that nothing else is calling it (as otherwise those would start failing silently)